source/common/router/config_impl.cc

```diff
@@ -1347,12 +1348,11 @@ RouteConstSharedPtr RouteEntryImplBase::clusterEntry(const Http::RequestHeaderMa
       return cluster_specifier_plugin_->route(shared_from_this(), headers);
     }
   }
-  return pickWeightedCluster(headers, random_value, true);
+  return pickWeightedCluster(headers, random_value);
 }
 
 RouteConstSharedPtr RouteEntryImplBase::pickWeightedCluster(const Http::HeaderMap& headers,
```

Comment:
Do you also want to add some documentation to this function definition, as Adi suggested https://github.com/envoyproxy/envoy/pull/34793#issuecomment-2259632156

Reply:
fixed